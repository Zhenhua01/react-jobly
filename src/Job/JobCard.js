/**
 * Presentational Component for Job
 * 
 * Prop: job {id, ...}
 * 
 * JobCardList -> Job Card
 */
function JobCard({ job }) {
  return (
    <div className="JobCard">
      <b>{job.title}</b>
      <p>Salary: {job.salary}</p>
      <p>Equity: {job.equity}</p>
    </div>

  );
}
export default JobCard;