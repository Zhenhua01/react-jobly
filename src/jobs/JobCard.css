.JobCard {
    color: black;
    box-sizing: border-box;
    padding: 1rem;
    text-align: left;
}
