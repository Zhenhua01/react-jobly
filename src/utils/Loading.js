import React from "react";
import "./Loading.css";


/** Loading message to display when making API calls for data. */

function Loading() {
  return (
    <div className="Loading">
      Loading ...
    </div>
  );
}

export default Loading;