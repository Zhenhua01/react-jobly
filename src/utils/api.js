import axios from "axios";

// Use port 3001 when connecting to local backend server
// const BASE_URL = process.env.REACT_APP_BASE_URL || "http://localhost:3001";

// Use heroku deployed backend when local server not available
const BASE_URL =
    process.env.REACT_APP_BASE_URL || "https://jobly-zhl-exp.herokuapp.com";

/** API Class.
 *
 * Static class tying together methods used to get/send to to the API.
 * There shouldn't be any frontend-specific stuff here, and there shouldn't
 * be any API-aware stuff elsewhere in the frontend.
 */

 class JoblyApi {

  static token = "";

  static async request(endpoint, data = {}, method = "get") {
    console.debug("API Call:", endpoint, data, method);

    const url = `${BASE_URL}/${endpoint}`;
    const headers = { Authorization: `Bearer ${JoblyApi.token}` };
    const params = (method === "get")
        ? data
        : {};

    try {
      return (await axios({ url, method, data, params, headers })).data;
    } catch (err) {
      console.error("API Error:", err.response);
      let message = err.response.data.error.message;
      throw Array.isArray(message) ? message : [message];
    }
  }

  // Individual API routes

  /** Get details on a company by handle. */

  static async getCompany(handle) {
    let res = await this.request(`companies/${handle}`);
    return res.company;
  }

  /** Get all companies or companies matching search term. */

  static async getCompanies(data = {}) {
    let res = await this.request(`companies`, data);
    return res.companies;
  }

  /** Get all jobs or jobs matching search term. */

  static async getJobs(data = {}) {
    let res = await this.request(`jobs`, data);
    return res.jobs;
  }

  /** Registers a new user. */

  static async signup(data) {
    let res = await this.request(`auth/register`, data, "post");
    return res.token;
  }

  /** Logins user. */

  static async login(data) {
    let res = await this.request(`auth/token`, data, "post");
    return res.token;
  }

  /** Update user information. */

  static async update(username, data) {
    let res = await this.request(`users/${username}`, data, "patch");
    return res.user;
  }

  /**Get user information. */

  static async getUserInfo(username) {
    let res = await this.request(`users/${username}`);
    return res.user;
  }

  /**Add job to user applications. */

  static async applyToJob(username, id) {
    await this.request(`users/${username}/jobs/${id}`, {}, "post");
    let res = await this.request(`users/${username}`);
    return res.user;
  }
}


export default JoblyApi;