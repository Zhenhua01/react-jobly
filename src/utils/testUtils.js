import React from "react";
import userContext from "../context/userContext";

const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  photo_url: null,
  applications: [],
};

const UserProvider = ({
  children,
  user = demoUser,
  hasAppliedToJob = () => false,
}) => (
  <userContext.Provider value={{ user, hasAppliedToJob }}>
    {children}
  </userContext.Provider>
);

export { UserProvider };
