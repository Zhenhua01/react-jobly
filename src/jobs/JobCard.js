import './JobCard.css';
import { useContext } from "react";
import userContext from '../context/userContext';
/**
 * Presentational Component for Job
 *
 * Prop: job {id, ...}
 *
 * JobCardList -> Job Card
 */
function JobCard({ job }) {
  const { user, apply } = useContext(userContext);
  const { id, title, salary, equity, companyName } = job;

  const hasApplied = user.applications.includes(id);


  return (
    <div className="JobCard container-fluid col-8 mb-3 bg-white rounded">
      <h5><u>{title}</u></h5>
      <p>
        {companyName}
        <br></br>
        Salary: {salary}
        <br></br>
        Equity: {equity}
      </p>

      <button
        onClick={() => { apply(id); }}
        disabled={hasApplied}
        className='btn btn-primary'>{hasApplied ? "Applied" : "Apply"}
      </button>

    </div>
  );
}
export default JobCard;