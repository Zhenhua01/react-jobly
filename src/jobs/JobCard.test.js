import React from "react";
import { render } from "@testing-library/react";
import JobCard from "./JobCard";
import { UserProvider } from "../utils/testUtils";


it("matches snapshot", function () {
  let job = { title: "CEO", salary: 1000000, equity: 10 };
  const { asFragment } = render(
      <UserProvider>
        <JobCard job={job} />
      </UserProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
