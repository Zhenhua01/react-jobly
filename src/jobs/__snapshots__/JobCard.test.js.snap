// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="JobCard container-fluid col-8 mb-3 bg-white rounded"
  >
    <h5>
      <u>
        CEO
      </u>
    </h5>
    <p>
      <br />
      Salary: 1000000
      <br />
      Equity: 10
    </p>
    <button
      class="btn btn-primary"
    >
      Apply
    </button>
  </div>
</DocumentFragment>
`;
